pkg/appgw/backendhttpsettings.go

```diff
@@ -274,5 +273,11 @@ func (c *appGwConfigBuilder) generateHTTPSettings(backendID backendIdentifier, p
 		httpSettings.RequestTimeout = to.Int32Ptr(reqTimeout)
 	}
 
+	if backendProtocol, err := annotations.BackendProtocol(backendID.Ingress); err == nil {
+		if backendProtocol == annotations.HTTPS {
+			httpSettings.Protocol = n.HTTPS
+		}
+	}
```

Comment:
It seems to me that if I annotated my Ingress incorrectly, I would not be told so by the software. It would be good to `glog` the error here (and perhaps `continue`).

Reply:
along with glog, it might help to add an event on ingress as well.